plugins/chain_plugin/CMakeLists.txt

```diff
@@ -4,7 +4,7 @@ add_library( chain_plugin
              chain_plugin.cpp
              ${HEADERS} )
 
-target_link_libraries( chain_plugin eosio_chain appbase )
-target_include_directories( chain_plugin PUBLIC "${CMAKE_CURRENT_SOURCE_DIR}/include" "${CMAKE_CURRENT_SOURCE_DIR}/../chain_interface/include" "${CMAKE_CURRENT_SOURCE_DIR}/../../libraries/appbase/include")
+target_link_libraries( chain_plugin producer_plugin eosio_chain appbase )
```

Comment:
I had no idea that would work:
https://cmake.org/cmake/help/latest/command/target_link_libraries.html#cyclic-dependencies-of-static-libraries

However, I'm not sure if we should do this or use the method providers of plugin_interface. I'm personally not a huge fan of the plugin_interface but that is the current way this is done. Thoughts @larryk85

Reply:
I will look into how to use plugin_interface.

Reply:
Do you know an existing code example? I saw register and subscribe of method providers. But it is immediately clear to me how to use it.